File: Java/Medium/Accepted/maximum-subarray-sum-with-one-deletion_1.java
class Solution {
    public int maximumSum(int[] arr) {
        
        int n = arr.length;
        int[] f = new int[n];
        int[] g = new int[n];
        f[0] = arr[0];
        g[0] = arr[0];
        for (int i = 1; i < n; i++) {
            f[i] = Math.max(f[i - 1] + arr[i], arr[i]);
            g[i] = Math.max(f[i - 1], g[i - 1] + arr[i]);
        }
        int ans = arr[0];
        for (int i = 0; i < n; i++) {
            ans = Math.max(ans, Math.max(f[i], g[i]));
        }
        return ans;
    }
}

File: Rust/Medium/WrongAnswer/maximum-subarray-sum-with-one-deletion_1.rs
impl Solution {
    pub fn maximum_sum(arr: Vec<i32>) -> i32 {
        
        let mut dp = vec![0; arr.len()];
        let mut dp2 = vec![0; arr.len()];
        dp[0] = arr[0];
        dp2[0] = arr[0];
        for i in 1..arr.len() {
            dp[i] = arr[i] + dp[i-1].max(0);
            dp2[i] = arr[i] + dp2[i-1].max(0);
            if arr[i] > dp2[i] {
                dp2[i] = arr[i];
            }
        }
        dp.iter().cloned().max().unwrap().max(dp2.iter().cloned().max().unwrap())
    }
}